### PR TITLE
TECH-4526: start up with Rails 5

### PR DESCRIPTION
**Pull request by ColinDKelley**
- Bumped `ActiveSupport` to `>= 5.2.4.3`.
- Removed dependence on `HoboSupport` in favor of `Invoca::Utils` and `ActiveSupport` and Ruby `&.`. For weird methods we didn't bother bringing across (like `rest`, `.try.`, `not_in?`, and `meta_def`), I just implemented them inline instead. Similarly I copied in `HoboSupport::Command` and `HoboSupport::Module`...and deleted all that we didn't need, which was most of it.
- Replaced `BlankSlate` with `ActiveSupport::ProxyObject`.
- Removed `alias_method_chain` in favor of `prepend`. And made an explicit `Model.mix_in` method to obviate all the `included` magic.
- Ran `magic_frozen_string_literal` to put that comment in every file.

There are no unit tests in this repo. :facepalm: But I did test with `pre_deploy_checker` and got to where `bundle install` and running tests there on models with `fields do` is working fine. (I didn't try running a db migration since that is out of scope.)

### CODE REVIEW

File: hobo_fields.gemspec [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name = File.basename( __FILE__, '.gemspec' )
-version = File.read(File.expand_path('../VERSION', __FILE__)).strip
+version = File.read(File.expand_path('VERSION', __dir__)).strip
```
Comment (jebentier) [inline hobo_fields.gemspec:2]: do we also need to update the gemspec to push the gem to gemfury for the major version bump that we're doing?

File: lib/hobo_fields.rb [hunk 2]
```diff
@@ -43,7 +47,7 @@ module HoboFields
   attr_reader :field_types
 
   def to_class(type)
-    if type.is_one_of?(Symbol, String)
+    if type.is_a?(Symbol) || type.is_a?(String)
```
Comment (jebentier) [inline lib/hobo_fields.rb]: can't this be simplified to a simple case statement?
```
case type
when Symbol, String
  ...
else
  ...
end
```

**Review by jebentier — COMMENTED** [review]
Just a couple nits for pushing the new version of the gem. Holy crap this is sketchy without having unit tests to test anything.

File: hobo_fields.gemspec (hunk 1 above)
Comment (ColinDKelley) [reply hobo_fields.gemspec:2]: Yes, I'm doing that next.

File: lib/hobo_fields.rb (hunk 2 above)
Comment (ColinDKelley) [reply lib/hobo_fields.rb]: Sure, that's a good idea!

**Comment by ColinDKelley** [comment]
> Holy crap this is sketchy without having unit tests to test anything.

Yeah. 🤦‍♂️

File: lib/hobo_fields.rb (hunk 2 above)
Comment (ColinDKelley) [reply lib/hobo_fields.rb]: Changed here: https://github.com/Invoca/hobo_fields/pull/16/commits/dfc1ce2a5b5af49bf03218e87a0d204fe7fe5b8a
No tests broke. :P

**Comment by ColinDKelley** [comment]
@jebentier Actually there is a test of `rdoctests` that I hadn't noticed. I'll try to get them to run.

**Comment by ColinDKelley** [comment]
@jebentier I got tests to run and fixed at least half of them. Go figure...they were a huge help! Now all the basics are working so we're actually pretty close to being able to generate migrations in Rails 5. There's still plenty of work left (like wiring in to Rails 5 foreign key support) but I'm going to stop here.

Changes since you saw it last:
```
    TECH-4526: move README to markdown and document how to run tests
    TECH-4526: remove .try.; Rails 5 fixes; skip foreign keys on SQLite
    TECH-4526: rename to belongs_to now that we're using prepend rather than alias_method_chain
    TECH-4526: fix private now that it's enforced in class << self
    TECH-4526: update tests that broke when we made :limit required and null: false the default
    TECH-4526: add .html_safe since gsub strips that away
    TECH-4526: add missing email Regexp
    TECH-4526: require gems needed for testapp
    TECH-4526: add gems back that are needed for the testapp
    TECH-4526: copy in .gitignore
    TECH-4526: add rubydoctest back to Gemfile
```

**Review by jebentier — APPROVED** [review]

**Comment by ColinDKelley** [comment]
I'm "merging" (moving) into a new `rails-5` feature branch.

**Pull request closed without merging**